sklearn/decomposition/pca.py

```diff
@@ -362,7 +362,7 @@ def _fit(self, X):
             raise TypeError('PCA does not support sparse input. See '
                             'TruncatedSVD for a possible alternative.')
 
-        X = check_array(X, dtype=[np.float64], ensure_2d=True,
+        X = check_array(X, dtype=[np.float64, np.float32], ensure_2d=True,
```

Comment:
@raghavrv this line makes it such that if `X` is integer based, it would be converted to  `np.float64` (the first element of the list) on all the platforms.

This answers your question at: https://github.com/scikit-learn/scikit-learn/pull/9067/files#r121114986